### PR TITLE
Rating feature and bug fixes

### PR DESCRIPTION
**Pull request by kristupas-g**

### CODE REVIEW

File: WebApi/Program.cs [hunk 1]
```diff
@@ -13,27 +14,12 @@ public class Program
...
-                await context.Database.MigrateAsync();
-                await ItemSeed.SeedData(context);
-            }
-            catch (Exception exception)
-            {
-                var logger = services.GetRequiredService<ILogger<Program>>();
-                logger.LogError(exception, "An error occured during migration");
-            }
-           
-            await host.RunAsync();
```
Comment (aauga) [inline WebApi/Program.cs:32]: why was this removed?

File: WebApi/Properties/launchSettings.json [hunk 2]
```diff
@@ -21,7 +22,7 @@
       "launchBrowser": true,
       "applicationUrl": "https://localhost:5001;http://localhost:5000",
       "environmentVariables": {
-        "ASPNETCORE_ENVIRONMENT": "Development"
+        "ASPNETCORE_ENVIRONMENT": "User"
```
Comment (aauga) [inline WebApi/Properties/launchSettings.json:25]: why was this changed?

File: WebApi/Program.cs (hunk 1 above)
Comment (kristupas-g) [reply WebApi/Program.cs:32]: Because we do our migrations manually eitherway

File: WebApi/Controllers/ItemsController.cs [hunk 3]
```diff
@@ -67,62 +67,62 @@ public async Task<IActionResult> CreateItem([FromForm] Item item)
...
+        [HttpPut("unsuspend/{id}")]
+        public async Task<IActionResult> UnsuspendItem([FromRoute] Guid id)
         {
             var user = await GetUser();
 
-            await Mediator.Send(new Redeem.Command { Id = id, User = user });
-            
+            await Mediator.Send(new Edit.Command { Id = id , User = user});
+
             return NoContent();
```
Comment (aauga) [inline WebApi/Controllers/ItemsController.cs:110]: what's the difference between edititem and unsuspenditem?

File: Persistence/Migrations/ItemDbContextModelSnapshot.cs [hunk 4]
```diff
@@ -58,6 +58,9 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.Property<DateTime>("ExpirationDate")
                         .HasColumnType("datetime2");
 
+                    b.Property<bool>("IsRecieved")
```
Comment (aauga) [inline Persistence/Migrations/ItemDbContextModelSnapshot.cs:61]: should be IsReceived

File: Application/Items/Commands/UnsuspendItem.cs [hunk 5]
```diff
@@ -30,14 +30,19 @@ public Handler(ItemDbContext context)
...
+              
                 if (item == null)
                 {
                     throw new NotFoundException(nameof(Item), request.Id);
                 }
+                if (item.Uploader != request.User)
+                {
+                    throw new ConflictException($"User {request.User} is not items uploader");
+                }
+                if (!item.IsRecieved)
```
Comment (aauga) [inline Application/Items/Commands/UnsuspendItem.cs:42]: should be IsReceived

File: Application/Items/Commands/UnsuspendItem.cs [hunk 6]
```diff
@@ -30,14 +30,19 @@ public Handler(ItemDbContext context)
...
+                {
+                    throw new ConflictException($"Item {item.Id} is already recieved");
```
Comment (aauga) [inline Application/Items/Commands/UnsuspendItem.cs:44]: should be received

File: WebApi/Controllers/ItemsController.cs (hunk 3 above)
Comment (aauga) [reply WebApi/Controllers/ItemsController.cs:110]: or you forgot to change it to unsuspenditem.command

File: Application/Items/Commands/Edit.cs [hunk 7]
```diff
@@ -40,7 +40,14 @@ public async Task<Unit> Handle(Command request, CancellationToken cancellationTo
                 {
                     throw new NotFoundException(nameof(Item), request.Id);
                 }
-
+                if(item.IsRecieved)
```
Comment (aauga) [inline Application/Items/Commands/Edit.cs:43]: change to isreceived

File: Application/Items/Commands/Edit.cs [hunk 8]
```diff
@@ -40,7 +40,14 @@ public async Task<Unit> Handle(Command request, CancellationToken cancellationTo
...
+                {
+                    throw new ConflictException($"Item {request.Id} has already been recieved");
```
Comment (aauga) [inline Application/Items/Commands/Edit.cs:45]: change to received

File: Application/Items/Commands/Delete.cs [hunk 9]
```diff
@@ -34,7 +34,7 @@ public async Task<Unit> Handle(Command request, CancellationToken cancellationTo
 
                 if (item == null)
                 {
-                    throw new NotFoundException(nameof(Item), request.Id);
+                    throw new NotFoundException($"Item {request.Id} does not exist");
```
Comment (aauga) [inline Application/Items/Commands/Delete.cs:37]: there's no point in changing this, as the 404 response already specifies that something does not exist

Comment (aauga) [inline Application/Items/Commands/Delete.cs:37]: this change isn't really necessary, as it gives the same response

File: Application/Items/Commands/ConfirmRecieving.cs [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    throw new NotFoundException($"Item {request.Id} does not exist");
+                }
+                if (user == null)
+                {
+                    throw new NotFoundException($"User {request.User} does not exist");
+                }
+                if(item.IsRecieved)
+                {
+                    throw new ConflictException($"Item {request.Id} has already been recieved");
```
Comment (aauga) [inline Application/Items/Commands/ConfirmRecieving.cs:52]: change to received

File: Application/Items/Commands/ChooseWinner.cs [hunk 11]
```diff
@@ -37,22 +37,44 @@ public async Task<Unit> Handle(Command request, CancellationToken cancellationTo
...
+                    throw new ConflictException($"Item {request.ItemId} has already been recieved");
+                }
+                if (item.Redeemer != null)
                 {
-                    ///Throw exception such contributor doesnt exist
+                    throw new ConflictException($"Item {request.ItemId} is suspended");
                 }
+                if(winner.User == request.User)
+                {
+                    throw new ConflictException($"Cant win your own item");
```
Comment (aauga) [inline Application/Items/Commands/ChooseWinner.cs:64]: message a bit harsh, change it to "You can not win your own item"

File: WebApi/Middleware/ErrorHandlerMiddleware.cs [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+                //LogErrorExceptionWithRequestBody(context, error);
+            }
+        }
+
+        private void LogErrorExceptionWithRequestBody(HttpContext context, Exception exception)
+        {
+            _logger.LogError($"Exception thrown: {exception} Method: {context.Request.Method}, Content: {context.Request.GetDisplayUrl()} Body: {JsonSerializer.Serialize(context.Request.Body).ToString()}");
+        }
+    }
+}
```
Comment (aauga) [inline WebApi/Middleware/ErrorHandlerMiddleware.cs:64]: don't really see any advantages of using this instead of a filter

Comment (aauga) [reply WebApi/Middleware/ErrorHandlerMiddleware.cs:64]: i'd rather use the previous version

File: WebApi/Controllers/ItemsController.cs (hunk 3 above)
Comment (kristupas-g) [reply WebApi/Controllers/ItemsController.cs:110]: I forgot to change it

**Comment by aauga** [comment]
looks good right now, just that we need to add .txt files to the .gitignore
